packages/libs/src/sdk/examples/react/src/App.tsx

```diff
@@ -0,0 +1,256 @@
+        scope: 'write'
+      })
+    }
+  }, [])
+
+  /**
+   * Fetch tracks based on the user handle present in handleInputRef
+   * Update favorites state with the results of the fetch
+   */
+  const fetchTrack = async () => {
+    const { data: selectedUser } = await audiusSdk.users.getUserByHandle({
+      handle: handleInputRef.current?.value ?? ''
+    })
+
+    const { data: tracks } = await audiusSdk.full.users.getTracksByUser({
+      id: selectedUser?.id ?? '',
+      userId: user?.userId ?? ''
+    })
+
+    setTracks(tracks ?? [])
+
+    const trackFavorites = (tracks ?? []).reduce<Record<string, boolean>>(
+      (result, track) => ({
+        ...result,
+        [track.id]: track.hasCurrentUserSaved
+      }),
+      {}
+    )
+
+    setFavorites(trackFavorites)
+  }
+
+  /**
+   * Set the streamUrl for the audio player based on the clicked track
+   */
+  const streamTrack = async (trackId: string) => {
+    if (trackId === playingTrackId) {
+      setIsPlaying((prev) => !prev)
+    } else {
+      setPlayingTrackSrc(await audiusSdk.tracks.streamTrack({ trackId }))
+      setPlayingTrackId(trackId)
+      setIsPlaying(true)
+    }
+  }
+
+  /**
+   * Favorite a track. This requires a user to be authenticated and granted
+   * write permissions to the app
+   */
+  const favoriteTrack =
+    (trackId: string): MouseEventHandler<HTMLButtonElement> =>
+    async (e) => {
+      e.stopPropagation()
+      if (user) {
+        setFavorites((prev) => ({ ...prev, [trackId]: true }))
+        try {
+          await audiusSdk.tracks.favoriteTrack({ userId: user.userId, trackId })
+        } catch (e) {
+          console.error('Failed to favorite track', e)
+          setFavorites((prev) => ({ ...prev, [trackId]: false }))
+        }
+      } else {
+        alert('Please log in with Audius to perform write operations')
+      }
+    }
+
+  /**
+   * Update the audio player based on the isPlaying state
+   */
+  useEffect(() => {
+    if (isPlaying && audioRef.current?.src) {
+      audioRef.current?.play()
+    } else {
+      audioRef.current?.pause()
+    }
+  }, [isPlaying])
+
+  return (
+    <HarmonyThemeProvider theme='day'>
+      <Flex direction='column' gap='m' m='2xl' alignItems='center'>
+        <Flex direction='column'>
+          <Text color='heading' strength='strong' variant='display'>
+            React + @audius/sdk
+          </Text>
+          <Text color='accent' variant='heading'>
+            Stream and favorite tracks!
+          </Text>
+        </Flex>
+        {!user ? (
+          <Hint
+            icon={() => <IconInfo size='l' color='default' />}
+            m='m'
+            css={{ maxWidth: 400 }}
+          >
+            <Flex gap='m' direction='column'>
+              <Text>
+                To perform writes with @audius/sdk please authorize this app to
+                perform writes on your behalf
+              </Text>
+              <div ref={loginWithAudiusButtonRef} />
```

Comment:
is this a button, or more of a container? just curious of the pattern here

Reply:
The oauth `renderButton` method just needs a ref to an html element to replace. So this does end up being a button